xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/vue/panels/LiveDataAdvancedPanelExtension.vue

```diff
@@ -0,0 +1,99 @@
+<!--
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<!--
+  The LiveDataAdvancedPanelExtension component is used to provide
+  extensions a way to inject a custom panel.
+-->
+<template>
+  <!--
+    Uses the LivedataBaseAdvancedPanel as root element, as it handles for us
+    all the Advanced Panels default behavior
+  -->
+  <LivedataBaseAdvancedPanel
+      :class="['livedata-advanced-panel-extension', 'livedata-advanced-panel-' + extension.id]"
+      :panel-id="extension.id"
+      ref="basePanel"
+  >
+
+    <!-- Provide the panel title and icon to the `header` slot -->
+    <template #header>
+      <XWikiIcon :icon-descriptor="{name: extension.icon}"/>
+      {{ extension.title }}
+    </template>
+
+    <!-- Define panel content inside the `body` slot -->
+    <template #body>
+      <div ref="bodyContainer" class="extension-body"></div>
+    </template>
+  </LivedataBaseAdvancedPanel>
+
+</template>
+
+<script>
+import LivedataBaseAdvancedPanel from "./LivedataBaseAdvancedPanel.vue";
+import XWikiIcon from "../utilities/XWikiIcon";
+
+export default {
+  name: "LiveDataAdvancedPanelExtension",
+
+  components: {
+    XWikiIcon,
+    LivedataBaseAdvancedPanel,
+  },
+
+  inject: ["logic"],
+
+  props: {'extension': Object},
+
+  methods: {
+    attachBodyElement()
+    {
+      // Make sure nothing happens if this method is called several times or at the wrong time.
+      if ('bodyContainer' in this.$refs) {
+        const bodyContainer = this.$refs.bodyContainer;
+        if (!bodyContainer.hasChildNodes()) {
+          bodyContainer.appendChild(this.extension.bodyElement);
```

Comment:
I find ``bodyElement`` misleading. I'm used to ``container`` or ``wrapper``, or even ``root``. Is there an ``extension.headElement``?

Reply:
I agree, I'll rename it to `container`. There is no `headElement` as I didn't want to let the extension provide the icon HTML code which could have lead to inconsistencies. In fact, all of the existing panels use the `#header` slot just for an icon and a string, it might make sense to move that template into the `LivedataBaseAdvancedPanel`, in particular if we want to make them more similar to the extensions and provide their data already in a `panels` array (I'm not so sure if this could work with the translations, though).